client/src/models/populate_services.js

```diff
@@ -7,322 +7,160 @@ import { lang } from "src/core/injected_build_constants.js";
-        };
-    }
-  })();
-
-  if (is_loaded) {
-    return Promise.resolve();
-  }
-
-  const time_at_request = Date.now();
+export const prefetch_services = (id) => {
   const client = get_client();
-  return client
-    .query({
-      query,
-      variables: {
-        lang: lang,
-        id,
-        _query_name: "services",
-      },
-    })
-    .then((response) => {
-      const response_data = response_data_accessor(response);
-
-      const resp_time = Date.now() - time_at_request;
-      if (!_.isEmpty(response_data)) {
-        // Not a very good test, might report success with unexpected data... ah well, that's the API's job to test!
-        log_standard_event({
-          SUBAPP: window.location.hash.replace("#", ""),
-          MISC1: "API_QUERY_SUCCESS",
-          MISC2: `Services, took ${resp_time} ms`,
-        });
-      } else {
-        log_standard_event({
-          SUBAPP: window.location.hash.replace("#", ""),
-          MISC1: "API_QUERY_UNEXPECTED",
-          MISC2: `Services, took ${resp_time} ms`,
-        });
-      }
-
-      const {
-        services,
-        serviceStandards,
-      } = extract_flat_data_from_hierarchical_response(response_data);
-
-      if (!_.isEmpty(services)) {
-        _.each(services, (service) => Service.create_and_register(service));
-      }
-      if (!_.isEmpty(serviceStandards)) {
-        _.each(serviceStandards, (standard) =>
-          ServiceStandard.create_and_register(standard)
-        );
-      }
-
-      // Need to use _.setWith and pass Object as the customizer function to account for keys that may be numbers (e.g. dept id's)
-      // Just using _.set makes large empty arrays when using a number as an accessor in the target string, bleh
-      _.setWith(
-        _subject_ids_with_loaded_services,
-        `${level}.${id}`,
-        true,
-        Object
-      );
-
-      // side effect
-      _.setWith(
-        _subject_ids_with_loaded_services,
-        `${level}.${id}`,
-        _.isEmpty(services),
-        Object
-      );
+  client.query({
+    query: services_query({
+      id: id || "gov",
+      service_fragments: "",
+    }),
+    variables: {
+      lang,
+      id: id || "gov",
+      _query_name: "services",
+    },
+  });
+  console.log(client.cache.data.data);
+};
 
-      return Promise.resolve();
-    })
-    .catch(function (error) {
-      const resp_time = Date.now() - time_at_request;
-      log_standard_event({
-        SUBAPP: window.location.hash.replace("#", ""),
-        MISC1: "API_QUERY_FAILURE",
-        MISC2: `Services, took ${resp_time} ms - ${error.toString()}`,
-      });
-      throw error;
+export const fetchServices = (query_options) => {
+  const time_at_request = Date.now();
+  const { id } = query_options;
+  const is_gov = id === "gov";
+  const variables = {
+    lang,
+    id,
+  };
+
+  const query = services_query(query_options);
+  const res = useQuery(query, { variables });
+  const { loading, error, data } = res;
```

Comment:
fetchServices is a hook, right? It took me a while to understand that, it should be called something like `useServices` or `useServicesData`